packages/tao/src/shared/package-manager.ts

```diff
@@ -41,7 +41,7 @@ export function getPackageManagerCommand(
 ): PackageManagerCommands {
   const commands: { [pm in PackageManager]: () => PackageManagerCommands } = {
     yarn: () => ({
-      install: 'yarn',
+      install: 'yarn dlx touch yarn.lock && yarn install',
       add: 'yarn add -W',
       addDev: 'yarn add -D -W',
```

Comment:
`-W` no longer used in the v3 version, so it should be : `yarn add -D`

Reply:
at the >13.9.0 version,the `packages/tao/src/shared/package-manager.ts` change to `packages/nx/src/shared/package-manager.ts`.

and add a new parameter:`addGlobal: 'yarn global add'` at the >13.9.0 version

.`yarn global`no longer used in the v3 version:[use-yarn-dlx-instead-of-yarn-global](https://yarnpkg.com/getting-started/migration#use-yarn-dlx-instead-of-yarn-global)